draft_order.go

```diff
@@ -0,0 +1,215 @@
+package goshopify
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/shopspring/decimal"
+)
+
+const draftOrdersBasePath = "admin/draft_orders"
+const draftOrdersResourceName = "draft_orders"
+
+// DraftOrderService is an interface for interfacing with the draft orders endpoints of
+// the Shopify API.
+// See: https://help.shopify.com/api/reference/orders/draftorder
+type DraftOrderService interface {
+	List(interface{}) ([]DraftOrder, error)
+	Count(interface{}) (int, error)
+	Get(int64, interface{}) (*DraftOrder, error)
+	Create(DraftOrder) (*DraftOrder, error)
+	Update(DraftOrder) (*DraftOrder, error)
+	Delete(int64) error
+	Invoice(int64, DraftOrderInvoice) (*DraftOrderInvoice, error)
+	Complete(int64, bool) (*DraftOrder, error)
+
+	// MetafieldsService used for DrafT Order resource to communicate with Metafields resource
+	MetafieldsService
+}
+
+// DraftOrderServiceOp handles communication with the draft order related methods of the
+// Shopify API.
+type DraftOrderServiceOp struct {
+	client *Client
+}
+
+// DraftOrder represents a shopify draft order
+type DraftOrder struct {
+	ID              int64            `json:"id,omitempty"`
+	OrderID         int64            `json:"order_id,omitempty"`
+	Name            string           `json:"name,omitempty"`
+	Customer        *Customer        `json:"customer,omitempty"`
+	ShippingAddress *Address         `json:"shipping_address,omitempty"`
+	BillingAddress  *Address         `json:"billing_address,omitempty"`
+	Note            string           `json:"note,omitempty"`
+	NoteAttributes  []NoteAttribute  `json:"note_attribute,omitempty"`
+	Email           string           `json:"email,omitempty"`
+	Currency        string           `json:"currency,omitempty"`
+	InvoiceSentAt   *time.Time       `json:"invoice_sent_at,omitempty"`
+	InvoiceURL      string           `json:"invoice_sent_at,omitempty"`
```

Comment:
InvoiceURL tag should be "invoice_url",  looks like it is repeating "invoice_sent_at"